llama-index-integrations/embeddings/llama-index-embeddings-yandexgpt/llama_index/embeddings/yandexgpt/base.py

```diff
@@ -0,0 +1,204 @@
+            "x-data-logging-enabled": "false",
+        }
+        try:
+            for attempt in Retrying(
+                stop=stop_after_attempt(self.retries),
+                wait=wait_fixed(self.sleep_interval),
+            ):
+                with attempt:
+                    response = requests.post(
+                        DEFAULT_YANDEXGPT_API_BASE, json=payload, headers=header
+                    )
+                    response = response.json()
+                    if "embedding" in response:
+                        return response["embedding"]
+                    raise YException(f"No embedding found, result returned: {response}")
+        except RetryError:
+            raise YException(
+                f"Error computing embeddings after {self.retries} retries. Result returned:\n{response}"
+            )
+
+    async def _aembed(self, text: str, is_document: bool = False) -> List[float]:
+        """
+        Embeds text using the YandexGPT Cloud API asynchronously.
+
+        Args:
+          text: The text to embed.
+          is_document: Whether the text is a document (True) or a query (False).
+
+        Returns:
+          A list of floats representing the embedding.
+
+        Raises:
+          YException: If an error occurs during embedding.
+        """
+        payload = {"modelUri": self._getModelUri(is_document), "text": text}
+        header = {
+            "Content-Type": "application/json",
+            "Authorization": f"Api-Key {self.api_key}",
+            "x-data-logging-enabled": "false",
+        }
+        try:
+            for attempt in Retrying(
+                stop=stop_after_attempt(self.retries),
+                wait=wait_fixed(self.sleep_interval),
+            ):
+                with attempt:
+                    async with aiohttp.ClientSession() as session:
+                        async with session.post(
+                            DEFAULT_YANDEXGPT_API_BASE, json=payload, headers=header
+                        ) as response:
+                            result = await response.json()
+                            if "embedding" in result:
+                                return result["embedding"]
+                            raise YException(
+                                f"No embedding found, result returned: {result}"
+                            )
+        except RetryError:
+            raise YException(
+                f"Error computing embeddings after {self.retries} retries. Result returned:\n{result}"
+            )
+
+    def _get_query_embedding(self, text: str) -> List[float]:
+        """Get query embedding sync."""
+        return self._embed(text, is_document=True)
+
+    def _get_query_embeddings(self, texts: List[str]) -> List[List[float]]:
+        """Get list of queries embeddings sync."""
+        embeddings = []
+        for text in texts:
+            embeddings.append(self._embed(text, is_document=True))
```

Comment:
same here?

Reply:
> same here?

Oh, I'm sorry, I didn't fully understand the difference between text and query. According to docs clarifai, embedding methods should be named vice versa. I have swapped the names of the embedding methods, please see if it's ok?